feat: Added label parameter to ui.menu

Pooja444 opened the pull request:
## Github Issue

https://github.com/h2oai/wave/issues/1577

## Description

Added a 'label' parameter to the UI component of wave. This label will be displayed beside the arrow for the menu dropdown. It will also be clickable, i.e. the dropdown will open on clicking the arrow as well as the label.

## Successful local test - 

<img width="514" alt="Screen Shot 2022-10-07 at 5 36 00 PM" src="https://user-images.githubusercontent.com/24620969/194678652-6dec2b16-8bf5-45a0-a148-dbbce5ee346e.png">

mturoci commented on ui/src/menu.tsx:
```diff
@@ -31,17 +31,19 @@ const css = stylesheet({
   /** An identifying name for this component. */
   name?: Id
+  /** The card's label. Mutually exclusive with the icon and image */
+  label?: S
```
Let's call this title instead of label.
```suggestion
  title?: S
```

mturoci commented on ui/src/menu.tsx:
```diff
@@ -31,17 +31,19 @@ const css = stylesheet({
   image?: S
   /** An identifying name for this component. */
   name?: Id
+  /** The card's label. Mutually exclusive with the icon and image */
```
```suggestion
  /** The text displayed next to the chevron. Mutually exclusive with the icon and image. */
```

mturoci commented on ui/src/menu.tsx:
```diff
@@ -31,17 +31,19 @@ const css = stylesheet({
   /** Commands to render. */
   items: Command[]
-  /** The card's icon. Mutually exclusive with the image. */
+  /** The card's icon. Mutually exclusive with the image and label. */
```
```suggestion
  /** The card's icon. Mutually exclusive with the image and title. */
```

mturoci commented on ui/src/menu.tsx:
```diff
@@ -31,17 +31,19 @@ const css = stylesheet({
+  /** The card's icon. Mutually exclusive with the image and label. */
   icon?: S
-  /** The card’s image, preferably user avatar. Mutually exclusive with the icon. */
+  /** The card’s image, preferably user avatar. Mutually exclusive with the icon and label. */
```
```suggestion
  /** The card’s image, preferably user avatar. Mutually exclusive with the icon and title. */
```

mturoci commented on ui/src/menu.tsx:
```diff
@@ -51,6 +53,7 @@ export const XMenu = ({ model }: { model: Menu }) => {
     <div data-test={name} className={clas(css.card, 'w-menu')} ref={ref} onClick={toggleMenu}>
       {image && <Fluent.Persona imageUrl={image} size={Fluent.PersonaSize.size48} styles={{ details: { padding: 0 } }} />}
       {icon && <Fluent.FontIcon className={css.icon} iconName={icon} />}
+      {label}
```
Let's add a bit of styling and themability.

```suggestion
      {title && <Fluent.Text variant='mediumPlus' styles={{ root: { color: cssVar('$text') } }}>{title}</Fluent.Text>}
```

mturoci reviewed (changes requested):
Looks great @Pooja444! A few minor changes. Also do not forget to run `make generate` after changing the docstrings/attr name. Could you also please add this newly added attribute into our `menu.py` example and also document it in `menu.md`?

Pooja444 commented:
@mturoci thanks for reviewing my PR! I addressed all your review comments, added an example for `title` in `menu.py`, added documented it in `menu.md`. I also ran `make generate` which automatically added `title` in a few other files. Please review and let me know if the updates look good.

I also tested it again after the changes - 

Successful local test -

<img width="798" alt="Screen Shot 2022-10-10 at 12 09 07 PM" src="https://user-images.githubusercontent.com/24620969/194936353-25fced80-9d7f-416f-b590-eb78b027e542.png">


Successful menu.py example run -

<img width="1135" alt="Screen Shot 2022-10-10 at 11 52 38 AM" src="https://user-images.githubusercontent.com/24620969/194936496-bd1a436a-1d56-4854-bf9d-a9e571591420.png">

<img width="1051" alt="Screen Shot 2022-10-10 at 12 09 50 PM" src="https://user-images.githubusercontent.com/24620969/194936529-38f2b942-4b22-4327-b919-c03d4ed8bed2.png">

mturoci reviewed (dismissed):
Thanks @Pooja444, this is one of the nicest community PRs we received so far! I will label it as `hacktoberfest-accepted`.

@lo5 can you please have a quick look at the API changes? Thanks!

Pooja444 commented:
> Looks great @Pooja444! A few minor changes. Also do not forget to run `make generate` after changing the docstrings/attr name. Could you also please add this newly added attribute into our `menu.py` example and also document it in `menu.md`?

Thanks a lot @mturoci !! 😁

Pooja444 commented:
> Thanks @Pooja444, this is one of the nicest community PRs we received so far! I will label it as `hacktoberfest-accepted`.
> 
> @lo5 can you please have a quick look at the API changes? Thanks!

Thanks a lot @mturoci !! 😁

Rmfjd1978 commented:
This change will be in wich h2o wave version?
bacause i've installed the  0.23.1 and the parameter label is not available on ui.menu

mturoci commented:
In the next release - 0.24. Currently available in nightly release.
